src/town_cmd.cpp

```diff
@@ -1445,6 +1445,9 @@ static void GrowTownInTile(TileIndex *tile_ptr, RoadBits cur_rb, DiagDirection t
 
 		if (!_settings_game.economy.allow_town_roads && !_generating_world) return;
 
+		/* Don't allow junctions on roadtypes which can't be built by towns. */
```

Comment:
Should both of these maybe be before the `_grow_town_result = GROWTH_SEARCH_STOPPED;`? After all, based on the comment it looks like it's definitely going to create some road, so it does not need to search further and then you still prevent it from building the road.
I've got no idea what the effect would be nor whether that would be a welcome effect.

Reply:
Not sure, there is a distinction between GROWTH_SUCCEED and GROWTH_SEARCH_STOPPED, but I don't know what.